app/uk/gov/hmrc/nationalinsurancerecord/services/NationalInsuranceRecordService.scala

```diff
@@ -68,78 +91,109 @@ class NationalInsuranceRecordService @Inject()(
+    desNIRecord: DesNIRecord,
+    desSummary: DesSummary,
+    desLiabilities: DesLiabilities,
+    manualCorrespondence: Boolean,
+    nino: Nino,
+    taxYear: TaxYear
+  ): Either[ExclusionResponse, NationalInsuranceTaxYear] = {
+    val purgedNIRecord: DesNIRecord =
+      desNIRecord.purge(desSummary.finalRelevantYear.get)
+
+    val exclusions: List[Exclusion] =
+      new DesExclusionService(
+        dateOfDeath              = desSummary.dateOfDeath,
+        liabilities              = desLiabilities.liabilities,
+        manualCorrespondenceOnly = manualCorrespondence
       ).getExclusions
 
-      if (exclusions.nonEmpty) {
-        metrics.exclusion(exclusions.head)
-        Left(ExclusionResponse(exclusions))
-      } else {
-        purgedNIRecord.niTaxYears.map(desTaxYearToNIRecordTaxYear).find(_.taxYear == taxYear.taxYear) match {
+    if (exclusions.nonEmpty) {
+      metrics.exclusion(exclusions.head)
+      Left(ExclusionResponse(exclusions))
+    } else {
+      purgedNIRecord
+        .niTaxYears
+        .map(desTaxYearToNIRecordTaxYear)
+        .find(_.taxYear == taxYear.taxYear) match {
           case Some(nationalInsuranceRecordTaxYear) =>
             Right(nationalInsuranceRecordTaxYear)
           case _ =>
             throw new NotFoundException(s"taxYear ${taxYear.taxYear} Not Found for $nino")
         }
-      }
     }
+  }
 
   private def desHomeResponsibilitiesProtection(liabilities: List[DesLiability]): Boolean =
     liabilities.exists(
       liability =>
-        NIRecordConstants.homeResponsibilitiesProtectionTypes.contains(liability.liabilityType.get)
+        homeResponsibilitiesProtectionTypes.contains(liability.liabilityType.get)
     )
 
   def calcPre75QualifyingYears(pre75Contributions: Int, dateOfEntry: Option[LocalDate], dateOfBirth: LocalDate): Option[Int] = {
     val yearCalc: BigDecimal = BigDecimal(pre75Contributions)/50
-    val sixteenthBirthday: LocalDate = dateOfBirth.plusYears(NIRecordConstants.niRecordMinAge)
-    val sixteenthBirthdayDiff: Int = NIRecordConstants.niRecordStart - TaxYearResolver.taxYearFor(sixteenthBirthday)
+    val sixteenthBirthday: LocalDate = dateOfBirth.plusYears(niRecordMinAge)
+    val sixteenthBirthdayDiff: Int = niRecordStart - TaxYearResolver.taxYearFor(sixteenthBirthday)
+
     val yearsPre75 = dateOfEntry match {
       case Some(doe) =>
-        (NIRecordConstants.niRecordStart - TaxYearResolver.taxYearFor(doe)).min(sixteenthBirthdayDiff)
+        (niRecordStart - TaxYearResolver.taxYearFor(doe)).min(sixteenthBirthdayDiff)
       case None =>
         sixteenthBirthdayDiff
     }
+
     if (yearsPre75 > 0) {
       Some(yearCalc.setScale(0, BigDecimal.RoundingMode.CEILING).min(yearsPre75).toInt)
     } else {
       None
     }
   }
 
-  private def desTaxYearToNIRecordTaxYear(desNITaxYear: DesNITaxYear): NationalInsuranceTaxYear = {
+  private def desTaxYearToNIRecordTaxYear(desNITaxYear: DesNITaxYear): NationalInsuranceTaxYear =
```

Comment:
What does this method do? It looks like these two case classes are almost identical.